src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunMetadataFieldLayoutAlgorithm.cs

```diff
@@ -729,10 +729,12 @@ protected override void AlignBaseOffsetIfNecessary(MetadataType type, ref Layout
             }
         }
 
-
         public static bool IsManagedSequentialType(TypeDesc type)
         {
-            type = type.UnderlyingType;
+            if (type is MetadataType metadataType && (metadataType.IsExplicitLayout || !metadataType.IsSequentialLayout))
```

Comment:
There are two places that call `IsManagedSequentialType`: one already has `type` as `MetadataType`, the other checks `IsValueType` first, which implies we can safely cast to `MetadataType` at the callsite. Maybe change the signature to take `MetadataType` instead of `TypeDesc`?

From that it will also be clear that the `type.Category == TypeFlags.Pointer` condition below is unreachable.

Reply:
Hmm, I tried to change the parameter type to MetadataType but that led to quite a bit of casting - after all, there are actually three places that call the method when you count its own recursion. I have instead slightly rearranged the method that I believe simplified it in general even without the type change.